### PR TITLE
Append option

### PR DESCRIPTION
**Pull request by jazzbee**
Requestor: Bartosz Jasinski
Risk: low 
Why: Needed for DATABASE_URL migration.
Usage:
--append '<var_name:suffix>'

example:
--append 'DATABASE_URL:?pool=10'

### CODE REVIEW

**Comment by kszarek** [comment]
I'm not sure that `addto` is the best name for it. Maybe `merge` or `merge-var` or ...

Tests need to be added before our review.

**Comment by ervamate** [comment]
@kszarek maybe `prepend`? Is this changing the stored variable?

**Comment by jmarmuszewski** [comment]
or `append` as we are in fact appending string to the variable value
or `mutate` - hard choice ;)

**Comment by kszarek** [comment]
@jazzbee readme needs to be updated

File: client/export_test.go [hunk 1]
```diff
@@ -147,6 +148,15 @@ func TestClient_ExportToTemplate(t *testing.T) {
...
 		},
+		{
+			name: "merged variable",
+			key:  "test/airmail/",
+			want: fmt.Sprintf("%s=%s\n%s=%s\n",
+				test.ShortKey4, test.KeyValueMap[test.Key4]+"?pool=10",
+				test.ShortKey5, test.KeyValueMap[test.Key5],
+			),
+			wantErr: false,
+		},
```
Comment (kszarek) [inline client/export_test.go]: Can we also test it with multiple variables case?

File: cmd/root.go [hunk 2]
```diff
@@ -50,6 +51,7 @@ func Execute() {
 func init() {
 	cobra.OnInitialize(initConfig)
 	RootCmd.PersistentFlags().StringVarP(&s3Region, "region", "r", "", "s3 region")
+	RootCmd.PersistentFlags().StringArrayVar(&append, "append", []string{}, "variable suffix, e.g: --addto \"DATABASE_URL:?pool=10\"")
```
Comment (kszarek) [inline cmd/root.go]: I'd like to avoid to add such global flags if not needed. Example:

```
/treasury import -h
...

Usage:
  treasury import ENVIRONMENT/APPLICATION/ secrets/file/path [flags]

Global Flags:
      --append stringArray   variable suffix, e.g: --addto "DATABASE_URL:?pool=10"

/treasury list -h
List the secrets set for a path

Usage:
  treasury list PATH [flags]

...

Global Flags:
      --append stringArray   variable suffix, e.g: --addto "DATABASE_URL:?pool=10"
```

The append flag is not used in the `import` or `list` commands.

File: client/export_test.go (hunk 1 above)
Comment (kszarek) [reply client/export_test.go]: Negative test case also would be recommended.

Comment (tzaleski) [reply client/export_test.go]: you might want to add test cases with empty key, empty value, 2+ `:`

File: client/export.go [hunk 3]
```diff
@@ -36,10 +37,23 @@ func (c *Client) Export(key, singleKeyExportFormat string) (string, error) {
 		sortedKeys = append(sortedKeys, secret.Key)
 		keySecretMap[secret.Key] = secret
 	}
+
+	var AppendMap map[string]string
+	AppendMap = make(map[string]string)
+	for _, val := range c.Append {
```
Comment (msamiczakAH) [inline client/export.go]: What if one would like to add `--append k1:v1 --append k1:v2` or `--append k1\:v1`? I think you need more sanitization here.

File: client/client.go [hunk 4]
```diff
@@ -16,6 +16,7 @@ type Client struct {
 
 	// Backend interface
 	Backend backend.BackendAPI
+	Append  []string
```
Comment (kszarek) [inline client/client.go]: There is no need to add `Append` here, on the Client level.
You can transfer the array as method args https://github.com/AirHelp/treasury/blob/6b7a6ce817c6f96f91c2b6f818c69b18c39503f7/cmd/template.go#L70

File: client/export.go [hunk 5]
```diff
@@ -36,10 +37,23 @@ func (c *Client) Export(key, singleKeyExportFormat string) (string, error) {
...
+		parts := strings.Split(val, ":")
+		if len(parts) == 2 {
+			AppendMap[parts[0]] = parts[1]
+		} else {
+			return "", errors.New("Bad append format (--append <variable>:<string>)")
```
Comment (kszarek) [inline client/export.go]: On the client level, I expect to get a proper data structure like `map[string]string`, not a string that we need to parse. Let's parse it on cmd package.
Be aware that the treasury Client package can be imported from other tools/packages e.g.: from Lambda Go Function.

File: test/bats/tests.bats [hunk 6]
```diff
@@ -123,6 +123,13 @@ invalid_aws_region=us-west-1
   [[ ${lines[0]} == "File with secrets successfully generated" ]]
 }
 
+@test "template-and-var-append" {
+  run $treasury template --src test/resources/bats-source.secret.tpl --dst test/output/bats-output.secret --append 'key1:treasury'
```
Comment (kszarek) [inline test/bats/tests.bats]: maybe we'll test something like `key1:?pool=10`

File: client/export_test.go [hunk 7]
```diff
@@ -147,6 +148,26 @@ func TestClient_ExportToTemplate(t *testing.T) {
 			),
 			wantErr: false,
 		},
+		{
+			name:   "merged variable",
+			key:    "test/airmail/",
+			append: []string{"DATABASE_URL:?pool=10"},
```
Comment (kszarek) [inline client/export_test.go]: Why can we not use `map[string]string` type as an arg to the method?

File: cmd/template.go [hunk 8]
```diff
@@ -55,14 +59,30 @@ func template(cmd *cobra.Command, args []string) error {
...
 	}
 
+	append, err := cmd.Flags().GetStringArray(templateCommandAppendArgument)
+	if err != nil {
+		return err
+	}
+
+	appendMap = make(map[string]string)
+	for _, val := range append {
+		parts := strings.Split(val, ":")
```
Comment (msamiczakAH) [inline cmd/template.go]: This part will fail if val will have multiple `:`. See https://github.com/AirHelp/treasury/blob/master/utils/utils.go#L62 for inspiration :trollface:
